autoPyTorch/optimizer/smbo.py

```diff
@@ -205,6 +219,28 @@ def __init__(self,
                                               port=self.logger_port)
         self.logger.info("initialised {}".format(self.__class__.__name__))
 
+        self.initial_configurations: typing.Optional[typing.List[Configuration]] = None
```

Comment:
Can we make L222-243 a separated function? (even not a function inside this class)
Then we can just test this function.

Reply:
sure